### PR TITLE
directive: pagination addition

### PR DESCRIPTION
**Pull request by drjova**

### CODE REVIEW

**Comment by lnielsen** [comment]
Just checked the PR on Zenodo, We also need to remove the "jump to end" of result.
screen shot 2016-02-05 at 15 25 49

Alternatively we need to provide the maximum number of results that you can get from ES but I think it's easier and better to just remove the possibility to jump to the end. I would still keep the "jump to start"

**Comment by drjova** [comment]
@lnielsen for completeness only shall we just create a template on the overlays without `jump to end`? It's a bit awkward to provide a pagination directive on search results without `jump to end` :/

File: src/invenio-search-js/invenioSearch.module.js [hunk 1]
```diff
@@ -779,13 +744,13 @@
         // Reset pages
         scope.paginatePages = [];
         // How many neighbours to show before and after the current page
-        var adjacent = 2;
+        var adjacent = 4;
```
Comment (lnielsen) [inline src/invenio-search-js/invenioSearch.module.js]: Can we parameterise this number so it can be configured by via the directive.

**Comment by lnielsen** [comment]
I don't see an easy way of getting rid of the two last pages by just using the template:

screen shot 2016-02-09 at 14 26 09

**Comment by jirikuncar** [comment]
What about showing only current range of pages +/- N as Google?
image
image

**Comment by lnielsen** [comment]
Yep, that's the idea. @drjova was just suggesting to do it via the template, and leave the code as is. (i.e being able to support both types). Not sure if it's needed to support both, but my preference is google-style so not sure if I"m a good candidate to take the decision :-) @tiborsimko do you have an opinion

**Comment by drjova** [comment]
@lnielsen I've added `adjacent-size (default: 4) ` and `show-go-to-first-last (default: false) ` parameters to the directive.

Example

```html
<invenio-search-pagination
  adjacent-size="4"
  show-go-to-first-last="true"
  template="/src/invenio-search-js/templates/pagination.html">
</invenio-search-pagination>
```

**Comment by lnielsen** [comment]
Ready to merge if you just update the commit message.

**Comment by drjova** [comment]
:shipit: ?

**Pull request merged**